components/setting_picture.tsx

```diff
@@ -93,49 +105,51 @@ export default class SettingPicture extends Component {
         }
     }
 
-    handleSave = (e) => {
+    handleSave = (e: React.MouseEvent) => {
         e.preventDefault();
         if (this.props.loadingPicture) {
             return;
         }
-        if (this.state.removeSrc) {
+        if (this.state.removeSrc && this.props.onRemove) {
             this.props.onRemove();
-        } else if (this.state.setDefaultSrc) {
+        } else if (this.state.setDefaultSrc && this.props.onSetDefault) {
             this.props.onSetDefault();
-        } else {
+        } else if (this.props.onSubmit) {
             this.props.onSubmit();
         }
     }
 
-    handleRemoveSrc = (e) => {
+    handleRemoveSrc = (e: React.MouseEvent) => {
         e.preventDefault();
         this.setState({removeSrc: true});
         this.focusFirstElement();
     }
 
-    handleSetDefaultSrc = (e) => {
+    handleSetDefaultSrc = (e: React.MouseEvent) => {
         e.preventDefault();
         this.setState({setDefaultSrc: true});
         this.focusFirstElement();
     }
 
-    handleFileChange = (e) => {
+    handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         this.setState({removeSrc: false, setDefaultSrc: false});
         this.props.onFileChange(e);
     }
 
     handleInputFile = () => {
-        this.selectInput.current.value = '';
-        this.selectInput.current.click();
+        if (this.selectInput && this.selectInput.current) {
```

Comment:
Checking `current` should be enough here
```suggestion
        if (this.selectInput.current) {
```